src/shared/dissect-image.c

```diff
@@ -231,6 +318,16 @@ static int dissected_image_probe_filesystems(DissectedImage *m, int fd) {
 
                 if (!p->rw)
                         p->growfs = false;
+
+                /* We might have learnt more about the device now (i.e. whether it is encrypted or not),
+                 * hence we need to validate this against policy again. */
+                found_flags = m->encrypted ?
+                        PARTITION_POLICY_ENCRYPTED :
+                        PARTITION_POLICY_UNPROTECTED|PARTITION_POLICY_VERITY|PARTITION_POLICY_SIGNED;
```

Comment:
you should read this as "everything but encrypted", i.e. we have the partition and can use it, but we know for sure it's not encrypted, but it could still be everything else, i.e. unprotected or with verity or even with verity sig.

the thing is this function only looks at individual partitions here, and if you do that you have no idea whether verity is available or not. You can only derive that from the whole set of partitions available, but that's something we already checked earlier when dissecting the partition table. Hence here, it's sufficent to validate a very "weak" flags mask again, whose only purpose is to say "it's definitely not encrypted" but nothing more. (after all image_policy_check_protection() checks *overlap* of the found mask passed in and the policy mask of the partition)

I'll add a comment about this.